scripts/findUnused.ts

```diff
@@ -1,32 +1,140 @@
+            }
+            return true
+        }
+    }
+    return false
+}
 
+const indexFiles = sourceFiles.filter(
+    (file) => file.getBaseName() === "index.ts"
+)
+for (const file of indexFiles) {
+    const publicApiExports = file
+        .getExportDeclarations()
+        .filter((declaration) => !exportAllRegex.test(declaration.getText()))
+        .map((declaration) => {
+            return {
+                path: relative(".", declaration.getSourceFile().getFilePath()),
+                exportedDeclarations: declaration
+                    .getNamedExports()
+                    .map((namedExport) => namedExport.getName())
+            }
+        })
+    if (publicApiExports.length) {
+        apiExports.push(...publicApiExports)
+    }
+}
 for (const sourceFile of project.getSourceFiles()) {
     const file = relative(".", sourceFile.getFilePath())
+    const sourceFileText = sourceFile.getFullText()
+    if (snippetFileRegex.test(file)) {
+        continue
+    }
     const unusedInFile = []
-    for (const exportedSymbol of sourceFile.getExportSymbols()) {
-        const exportName = exportedSymbol.getName()
-        const references = exportedSymbol
-            .getDeclarations()
-            .flatMap((declaration) => {
-                if (declaration.getKindName() === "ExportSpecifier") {
+    for (const [name, declarations] of sourceFile.getExportedDeclarations()) {
+        const references = declarations.flatMap((declaration) => {
+            if (declaration.getKindName() === "ExportSpecifier") {
+                return []
+            }
+            if (declaration.getSourceFile() !== sourceFile) {
+                return []
+            }
+            if (exportAllRenamedRegex.test(declaration.getText())) {
+                return []
+            }
+            const knownExport = apiExports.find(
+                (apiExport) => apiExport.path === file
+            )
+            if (knownExport) {
+                if (knownExport.exportedDeclarations.includes(name)) {
                     return []
                 }
-                if (declaration.getSourceFile() !== sourceFile) {
-                    return []
+            }
+            const references = (declaration as any as BindingNamedNode)
+                .findReferences()
+                .flatMap((ref) => ref.getReferences())
+
+            const sourceFileAndReferencesData = {
+                filePath: file,
+                sourceFileText,
+                referencesLength: references.length,
+                declarationName: (
+                    declaration as any as BindingNamedNode
+                ).getName()
+            }
+            switch (declaration.getKindName()) {
+                case "VariableDeclaration":
+                    {
+                        const foundComment = checkForIgnoreUnusedComment({
+                            ...sourceFileAndReferencesData,
+                            declarationNode: declaration
```

Comment:
Could we instead here use something like `getAncestorOfKindOrThrow` (I'm not sure that exists but I think it does)? It would look a little less fragile in case we haven't encountered some edge case where the declaration is not exactly two levels up.